### PR TITLE
[SDFAB-959] Add PFCP session struct to keep track of sessions

### PR DESCRIPTION
**Pull request by EmanueleGallone**
Adding a session struct eases keeping track of established sessions, allowing for more complex operations like session modification.

### CODE REVIEW

**Review by osinstom — COMMENTED** [review]
I would rename the file to `pkg/pfcpsim/session.go`

File: pkg/pfcpsim/pfcp_session.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	LocalSEID uint64
+	PeerSEID  uint64
+
+	PDRs []*ie.IE
+	FARs []*ie.IE
+
+	QERs []*ie.IE
+}
+
+func NewSession(localSEID uint64, peerSEID uint64) *PFCPSession {
```
[the remark this answers is not in the document]
Comment (EmanueleGallone) [reply pkg/pfcpsim/pfcp_session.go]: Actually no. We can set it later on

File: pkg/pfcpsim/session.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	LocalSEID uint64
+	PeerSEID  uint64
+
+	PDRs []*ie.IE
+	FARs []*ie.IE
+
+	QERs []*ie.IE
+}
+
+func NewSession() *PFCPSession {
```
Comment (osinstom) [inline pkg/pfcpsim/session.go:15]: I'm wondering if we should export `NewSession`. Should a user be able to create a session, without establishing it? I don't think so. 

Also, we should make fields of `PFCPSession` private IMO.

File: cmd/pfcpsim-client/main.go [hunk 3]
```diff
@@ -390,15 +400,21 @@ func createSessions(count int) {
...
-		log.Infof("Created session")
+		activeSessions = append(activeSessions, &PFCPClientContext{
+				session: sess,
+				pdrs:    pdrs,
+				fars:    fars,
+				qers:    qers,
+			},
+		)
+
+		log.Infof("Created new PFCP session")
```
Comment (EmanueleGallone) [inline cmd/pfcpsim-client/main.go:417]: having private fields, I cannot show session's SEID. is it ok?

File: pkg/pfcpsim/pfcp_session.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (EmanueleGallone) [reply pkg/pfcpsim/pfcp_session.go]: removed `NewSession` method. We can instantiate the object directly

File: pkg/pfcpsim/session.go (hunk 2 above)
Comment (EmanueleGallone) [reply pkg/pfcpsim/session.go]: made PFCPSession fields private. `NewSession` has been removed, as agreed offline

File: cmd/pfcpsim-client/main.go [hunk 4]
```diff
@@ -362,14 +372,14 @@ func createSessions(count int) {
...
+				WithDstInterface(ieLib.DstInterfaceAccess).
 				WithTEID(downlinkTEID).
 				WithDownlinkIP(nodeBAddress.String()).
 				BuildFAR(),
 		}
 
-		qers := []*ie.IE{
-			// session QER
+		qers := []*ieLib.IE{
+			// PFCPClientContext QER
```
Comment (osinstom) [inline cmd/pfcpsim-client/main.go:382]: perhaps replaced by IDE

File: cmd/pfcpsim-client/main.go (hunk 3 above)
Comment (osinstom) [reply cmd/pfcpsim-client/main.go:417]: it's ok

File: cmd/pfcpsim-client/main.go (hunk 4 above)
Comment (EmanueleGallone) [reply cmd/pfcpsim-client/main.go:382]: oh yeah. thanks for catching this

File: cmd/pfcpsim-client/main.go (hunk 3 above)
Comment (osinstom) [reply cmd/pfcpsim-client/main.go:417]: at least for now

**Review by osinstom — APPROVED** [review]
I left only minor comment, please address it and we can merge

**Pull request merged**